### PR TITLE
Update primitive-types crate

### PR DESCRIPTION
**Pull request by NikVolf**
`-` ancient rand dependency

### CODE REVIEW

**Comment by NikVolf** [comment]
Actually, it seems that it gives like ~20% performance hit during block import
Probably due to https://github.com/paritytech/parity-common/pull/317

And `primitive-types/libc` feature is explicitly used in `sp-core`

@ordian ^^^

**Comment by NikVolf** [comment]
with `primitive-types/libc`

```
import block/Wasm       time:   [4.5876 s 4.6357 s 4.7006 s]
import block/Native     time:   [2.5677 s 2.6154 s 2.6846 s]  
```

without `primitive-types/libc`:

```
import block/Wasm       time:   [4.7306 s 4.8279 s 4.9466 s]                               
                        change: [+3.1825% +5.5383% +7.6266%] (p = 0.00 < 0.05)
                        Performance has regressed.

import block/Native     time:   [2.7783 s 2.8249 s 2.8624 s]                                 
                        change: [+2.5561% +6.1032% +9.7396%] (p = 0.00 < 0.05)
                        Performance has regressed.

```

**Comment by ordian** [comment]
> Actually, it seems that it gives like ~5% performance hit during block import
> Probably due to [paritytech/parity-common#317](https://github.com/paritytech/parity-common/pull/317)
> 
> And `primitive-types/libc` feature is explicitly used in `sp-core`
> 
> @ordian ^^^

I opened https://github.com/rust-lang/rust/issues/69024. 

We haven't published a new version of `fixed-hash` and if you think the `libc` feature removal should be reverted, we still can do it.

**Comment by NikVolf** [comment]
> We haven't published a new version of `fixed-hash` and if you think the `libc` feature removal should be reverted, we still can do it.

Yeah, the fact that it is not published probably unblocks this pr.
Performance issues still should be addressed though - I've benched against just latest `parity-common`, but we'll figure out later.

File: primitives/core/Cargo.toml [hunk 1]
```diff
@@ -70,7 +70,6 @@ std = [
 	"primitive-types/serde",
 	"primitive-types/byteorder",
 	"primitive-types/rustc-hex",
-	"primitive-types/libc",
```
Comment (bkchr) [inline primitives/core/Cargo.toml]: Don't want we bring it back for now?

Comment (NikVolf) [reply primitives/core/Cargo.toml]: Yep

**Pull request merged**